static/app/components/notificationActions/forms/slackForm.tsx

```diff
@@ -0,0 +1,139 @@
+import {useMemo, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import DropdownButton from 'sentry/components/dropdownButton';
+import {DropdownMenu} from 'sentry/components/dropdownMenu';
+import Input from 'sentry/components/input';
+// import {
+//   NotificationActionCell,
+//   NotificationActionFormContainer,
+// } from 'sentry/components/notificationActions/notificationActionItem';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {
+  AvailableNotificationAction,
+  NotificationAction,
+} from 'sentry/types/notificationActions';
+
+type SlackFormProps = {
+  /**
+   * The notification action being represented
+   */
+  action: Partial<NotificationAction>;
+  /**
+   * The available actions for the action's serviceType (e.g. "slack", "pagerduty")
+   */
+  availableActions: AvailableNotificationAction[];
+  onCancel: () => void;
+  onChange: (name: string, value: any) => void;
+  onSave: () => void;
+};
+
+const SlackForm = ({
+  action,
+  availableActions,
+  onChange,
+  onSave,
+  onCancel,
+}: SlackFormProps) => {
+  // Maps integrationId to integrationName
+  const availableWorkspaces: Record<number, string> = useMemo(() => {
+    const workspacesMap: Record<number, string> = {};
+    availableActions.forEach(service => {
+      if (service.action.integrationId && service.action.integrationName) {
+        workspacesMap[service.action.integrationId] = service.action.integrationName;
+      }
+    });
+    return workspacesMap;
+  }, [availableActions]);
+
+  const [selectedWorkspace, setSelectedWorkspace] = useState(
```

Comment:
nit: doesn't matter much, but i prefer to bubble state's to the top, and keep all other constants listed below 🤷